Add "include" option, rename "compileOptions"

tyler-boyd opened the pull request:
Hey @EMH333, just opening this as a draft for now to get your feedback. Plz disregard the dist/ and gitignore changes; if you're interested in this PR I'll fix it up and remove them.

* Adds an `include` option, similar to the svelte rollup plugin's `include` option. This is necessary to support mixing/matching webcomponents. In general, we include the plugin twice. Once with `compilerOptions.customElement = true` and `include: /\.wc\.svelte$/`, and a second time without `customElement` and with `exclude: /\.wc\.svelte$/`. I couldn't figure out how to make `exclude` work so I only did `include`, and it works well enough for our production build 🙂 
* Renamed `compileOptions` to `compilerOptions` -- small thing, but this lets you import your `svelte.config.js` and just use that object instead of having to copy in both the compilerOptions and preprocess settings

Let me know what you think, thanks!

EMH333 commented:
> Hey @EMH333, just opening this as a draft for now to get your feedback. Plz disregard the dist/ and gitignore changes; if you're interested in this PR I'll fix it up and remove them.

Thanks for your PR! Definitely interested

> 
> Adds an `include` option, similar to the svelte rollup plugin's `include` option. This is necessary to support mixing/matching webcomponents. In general, we include the plugin twice. Once with `compilerOptions.customElement = true` and `include: /\.wc\.svelte$/`, and a second time without `customElement` and with `exclude: /\.wc\.svelte$/`. I couldn't figure out how to make `exclude` work so I only did `include`, and it works well enough for our production build :slightly_smiling_face:
> 

Looks good

>  Renamed `compileOptions` to `compilerOptions` -- small thing, but this lets you import your `svelte.config.js` and just use that object instead of having to copy in both the compilerOptions and preprocess settings
> 

I'd be inclined to also keep `compileOptions` around for at least another "major" version (this PR, and a few other things I'm working on, would kick us up to `0.6.0` and I'd want to allow but deprecate the old name till we get to `0.7.0`). This esbuild plugin tends to be a keystone of people's build code and I try to be conservative when making changes like this. Clear warnings are better than errors when upgrading. a24ef7c9915be61cdbc2d31ad8bc98731eff0ceb is where I renamed `preprocessor` to `preprocess`, also in a quest for compatibility. That seemed to work fairly well at the time, but I'm open to hearing your thoughts

EMH333 commented:
@tyler-boyd Any updates here?

tyler-boyd commented:
Thanks @EMH333 , I've updated the PR to be a bit more ready! Backwards compatible, and rebased/cleaned up history.

Please take a look when you can and let me know if there should be any changes!

EMH333 commented:
Looks great, I'll get this merged this weekend. Thank you!

EMH333 commented:
Released as a part of `v0.6.0` :tada:

tyler-boyd commented:
Thank you very much @EMH333! We're now pointing back at the real esbuild-svelte v0.6.0 instead of the fork 🙂
